src/optionsutils.nim

```diff
@@ -497,3 +497,37 @@ template optCmp*(self, cmp, value: untyped): untyped =
       if `cmp`(unsafeGet(a), unsafeGet(b)):
         return a
   )()
+
+import sequtils
+
+macro withSome*(body: untyped): untyped =
+  ##Early exit if the any of option parameters passed are none.
+  ##Also shadows the parameters to their internal type.
+  let 
+    identDefs = body[3] #All parameter names
+    stmtList = body.findChild(it.kind == nnkStmtList) #body
```

Comment:
These `findChild` statements should really be replaced with indices, or better yet use the named accessors from `macroutils`. This statement (with the above `body` -> `procDef` rename) could then be written as `procDef.body`.